tests/common/helpers/voq_lag.py

```diff
@@ -0,0 +1,341 @@
+    Args:
+        duthost<object>: duthost
+        asic<object>: asic object
+        portchannel_members<list> : portchannel members
+        portchannel_ip<str>: portchannel ip
+        portchannel<str> : portchannel
+        add<bool> : True adds portchannel
+    """
+    if add:
+        config_cmd = "config portchannel {}"\
+            .format(asic.cli_ns_option if asic.cli_ns_option else "")
+        duthost.shell("{} add {}".format(config_cmd, portchannel))
+        int_facts = duthost.interface_facts(namespace=asic.namespace)['ansible_facts']
+        pytest_assert(int_facts['ansible_interface_facts'][portchannel])
+
+    if portchannel_members:
+        for member in portchannel_members:
+            duthost.shell("config portchannel {} member add {} {}"
+                          .format(asic.cli_ns_option, portchannel, member))
+
+    if portchannel_ip:
+        duthost.shell("config interface {} ip add {} {}"
+                      .format(asic.cli_ns_option, portchannel, portchannel_ip))
+        int_facts = duthost.interface_facts(namespace=asic.namespace)['ansible_facts']
+        pytest_assert(int_facts['ansible_interface_facts']
+                      [portchannel]['ipv4']['address'] == portchannel_ip.split('/')[0])
+        time.sleep(30)
+        int_facts = duthost.interface_facts(namespace=asic.namespace)['ansible_facts']
+        pytest_assert(int_facts['ansible_interface_facts'][portchannel]['link'])
+
+
+def verify_lag_id_is_unique_in_chassis_db(duthosts, duthost, asic):
+    """
+    Verifies lag id is unique for a newly added LAG in CHASSIS_DB
+    args:
+        duthosts <list>: duthost
+        duthost <obj>: duthost
+        asic <obje>: asic
+
+    """
+    logging.info("Verifying on duthost {} asic {} that lag id is unique"
+                 .format(duthost.hostname, asic.asic_index))
+    lag_id_list = get_lag_ids_from_chassis_db(duthosts)
+    add_lag(duthost, asic)
+    added_pc_lag_id = get_lag_id_from_chassis_db(duthosts)
+    if added_pc_lag_id in lag_id_list:
+        pytest.fail('LAG id {} for newly added LAG {} already exist in lag_id_list {}'
+                    .format(added_pc_lag_id, TMP_PC, lag_id_list))
+
+    logging.info('LAG id {} for newly added LAG {} is unique.'
+                 .format(added_pc_lag_id, TMP_PC))
+
+
+def verify_lag_in_app_db(asic, deleted=False):
+    """
+    Verifies lag in ASIC APP DB.
+    It runs the command e.g. 'redis-cli -n 0 --raw keys "*LAG_TABLE*"'
+    Args:
+        asic<obj>: asic
+        deleted<bool>: False if lag is not deleted
+    """
+    appdb = AppDbCli(asic)
+    app_db_lag_list = appdb.get_app_db_lag_list()
+    if deleted:
+        for lag in app_db_lag_list:
+            if TMP_PC not in lag:
+                logging.info('LAG {} is deleted in ASIC app db'.format(TMP_PC))
+                return
+        pytest.fail('LAG {} still exist in ASIC app db,'
+                    ' Expected was should be deleted from asic app db.'.format(TMP_PC))
```

Comment:
Is there a logic issue here?
If `deleted` is `True`, then the current code would return if `TMP_PC` not in the first lag in `app_db_lag_list`. However, it is possible that `TMP_PC` in rest of the lag in `app_db_lag_list`. So, the correct logic should be like below, right?
```
    if deleted:
        for lag in app_db_lag_list:
            if TMP_PC in lag:
                pytest.fail('LAG {} still exist in ASIC app db,'
                            ' Expected was should be deleted from asic app db.'.format(TMP_PC))
        logging.info('LAG {} is deleted in ASIC app db'.format(TMP_PC))
        return

```